### PR TITLE
Minor profiler refactor to avoid string construction

### PR DESCRIPTION
**Pull request by zwxxx**
std string concatenation allocates. Check if profiler enabled then call EndTimeAndRecordEvent to avoid string construction and allocation. 
@pranavsharma
@tracysh

### CODE REVIEW

File: onnxruntime/core/common/profiler.cc [hunk 1]
```diff
@@ -33,15 +33,14 @@ void Profiler::StartProfiling(const std::string& file_name) {
 void Profiler::EndTimeAndRecordEvent(EventCategory category,
                                      const std::string& event_name,
                                      TimePoint& start_time,
-                                     const std::initializer_list<std::pair<std::string, std::string>>& event_args,
```
Comment (pranavsharma) [inline onnxruntime/core/common/profiler.cc]: Won't this take away the ability to record multiple key-val pairs in a profile? Isn't invoking this function conditionally (that is only when profiling is enabled) enough?

Comment (zwxxx) [reply onnxruntime/core/common/profiler.cc]: I dont see why one EndTimeAndRecordEvent should take multiple events. At least existing code has no such cases. But if you insist, I can add it back.

Comment (pranavsharma) [reply onnxruntime/core/common/profiler.cc]: Thinking more about it, yeah recording just one event is fine. However we should've a name for the thing that you're recording. So in this case I don't know if the string that you're recording is an op_name or something else. Would be good to add event_name and event_value parameters to the function.

Comment (zwxxx) [reply onnxruntime/core/common/profiler.cc]: ok. added std::pair<std::string, std::string> back.

File: onnxruntime/core/common/profiler.cc [hunk 2]
```diff
@@ -33,14 +33,14 @@ void Profiler::StartProfiling(const std::string& file_name) {
 void Profiler::EndTimeAndRecordEvent(EventCategory category,
                                      const std::string& event_name,
                                      TimePoint& start_time,
-                                     const std::string& op_name,
+                                     const std::pair<std::string, std::string>& op_name,
```
Comment (pranavsharma) [inline onnxruntime/core/common/profiler.cc]: Thanks for this change. This pair can contain stuff besides op_name too, right? what's a good name for this parameter? tag?

Comment (zwxxx) [reply onnxruntime/core/common/profiler.cc]: events?

Comment (zwxxx) [reply onnxruntime/core/common/profiler.cc]: renamed to tags.

**Comment by pranavsharma** [comment]
Looks like some of the tests failed. Can you rebase?

**Comment by yufenglee** [comment]
I would like to close this PR as it has been inactive for a while and similar change was merged.

**Pull request closed without merging**